enhancements/builds/multi-arch-builds.md

```diff
@@ -0,0 +1,456 @@
+* Ensure that the `openshift-controller-manager-images` configMap in the `openshift-controller-manager-operator` namespace references builder and deployer images using manifest lists and not architecture-specific images.
+* Add an optional target platforms []string to github.com/openshift/api/openshiftcontrolplane/v1.BuildOverridesConfig.
+* Add an optional "archNodeSelector" *bool to github.com/openshift/api/openshiftcontrolplane/v1.BuildOverridesConfig.
+* Add an optional "manifestList" *bool to github.com/openshift/api/openshiftcontrolplane/v1.BuildOverridesConfig.
+* Figure out where the platforms list for builds should come from if the configured BuildOverridesConfig doesn't have one.
+* Teach openshift-controller-manager to launch, monitor, and when necessary cancel the N+1 pods, setting the target platform and output location differently in the encoded Build values it passes to the various platform-specific build pods, unless "manifestList" is explicitly disabled.
+  * Build pods will gain a label which `oc logs` can use to select which of the build's pods it will read logs from.
+  * If a node with the corresponding OS and architecture selector labels (e.g. "linux", "amd64") is not able to build for any requested variant (e.g. "", "v2", etc.), more refinement will be required.
+  * Unless BuildOverridesConfig.archNodeSelector is explicitly disabled, each pod will have a "kubernetes.io/arch" value added to its "nodeSelector".
+* Update `oc logs` updated Build and BuildConfig types. Update log viewing UX.
+* Update `oc describe` for updated Build and BuildConfig types. Update build progress / status UX.
+* Extend `oc new-app` to check the architectures of the images that the BuildConfig it's about to generate uses.
+
+### Risks and Mitigations
+
+* This proposal doesn't grant new privileges to any of the components involved, but it should still get a security review.
+* Each pod will generate its own logs, so this will require updates to the UX to present them distinctly.
+* Registries which don't allow untagging: ignore error results; we're only untagging to try to be nice.
+* While we can dutifully mark a built image as e.g. linux/amd64 or linux/amd64/v2 when we're told to build one or the other, we're dependent on the build's Dockerfile and the commands it runs to actually produce binaries that are correct for the target platform variant.
+
+### Drawbacks
+
+* Limited to using architectures available on the cluster.
+* Builds which use base images which are not available for all of the cluster's architectures will require manual configuration, either of the build controller to limit the target platforms for all builds, or of individual affected Builds or BuildConfigs to limit the target platforms for them.
```

Comment:
so this would mean it would only limit builds to the base images' platform availability and if any of the platforms not supported are in the  required list, then it fails right away ?

Reply:
Yes, a builder will fail when it attempts to pull a base image for an architecture that the base image wasn't built for.  The build controller doesn't know which base images a build will need, or which architectures those images have been built for.  That combined with a "default to requiring builds to succeed on all available platforms" is a potential source of trouble when a new node architecture is added to a cluster.